### PR TITLE
Added an option to name the streams

### PR DESCRIPTION
**Pull request by njezersek**
> Please note that I am not very familiar with the codebase so if there is something I have missed I am open to changing the implementation or even closing the request.

### Naming the x out streams
Right now it seems to be very hard to know from which camera (left, right, rgb, ...) a packet came from.

Take a look at this example:
```python
from depthai_sdk import OakCamera

with OakCamera() as oak:
	color = oak.create_camera('color')
	left = oak.create_camera('left')
	right = oak.create_camera('right')
	oak.sync([color.out.main, left.out.main, right.out.main], callback=lambda p: print(p))
	oak.start(blocking=True)

# prints
# {'0_preview': <depthai_sdk.classes.packets.FramePacket object at 0x00000164DDCA3400>, '1_out': <depthai_sdk.classes.packets.FramePacket object at 0x00000164DDCA3310>, '2_out': <depthai_sdk.classes.packets.FramePacket object at 0x00000164DDCA3340>}
```

As you can see the frames are labeled with "0_preview", "1_out", "2_out" which is not very developer friendly. Ideally, I would like to have the streams named the same as the camera source ("color", "left", "right" in this example) but I am afraid this would break existing code so I'm proposing a middle ground solution (with minimal change to the SDK source) which allows the user to manually set the stream name when creating a camera.

Here is an example:
```python
from depthai_sdk import OakCamera

with OakCamera() as oak:
	color = oak.create_camera('color', name="color")
	left = oak.create_camera('left', name="left")
	right = oak.create_camera('right', name="right")
	oak.sync([color.out.main, left.out.main, right.out.main], callback=lambda p: print(p))
	oak.start(blocking=True)

# prints
# {'right': <depthai_sdk.classes.packets.FramePacket object at 0x0000012E3FCB80D0>, 'left': <depthai_sdk.classes.packets.FramePacket object at 0x0000012E3FCB89D0>, 'color': <depthai_sdk.classes.packets.FramePacket object at 0x0000012E3FCB8550>}
```

The frames are labeled as specified in the `create_camera` function.

### Fixed `setup.py`

I have also made a small change to the `setup.py`. The `requirements.txt` contains some flags (`--extra-index-url ...`) but the `setup` function expects just a list of requirements. So I added a filter to remove the flags. Please note that you have to provide these flags manually when installing the package like so:
```
pip install --extra-index-url https://artifacts.luxonis.com/artifactory/luxonis-python-snapshot-local/ -e ./depthai_sdk
```

### CODE REVIEW

File: depthai_sdk/src/depthai_sdk/oak_camera.py [hunk 1]
```diff
@@ -123,6 +124,7 @@ def create_camera(self,
                                encode=encode,
                                rotation=self._rotation,
                                replay=self.replay,
+                               name=name,
```
Comment (Erol444) [inline depthai_sdk/src/depthai_sdk/oak_camera.py:127]: Add the name to the comments underneath the method (both inside oak_camera and CameraComponent), so it gets auto-populated to the documentation. Also perhaps we could add this functionality to other components?

**Review by Erol444 — APPROVED** [review]

File: depthai_sdk/setup.py [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import io
 
 from setuptools import setup
+import pkg_resources
```
Comment (Erol444) [inline depthai_sdk/setup.py:4]: Leave this out perhaps?

**Review by Erol444 — APPROVED** [review]

**Review by daniilpastukhov — CHANGES_REQUESTED** [review]
The encoded output isn't named correctly in the following example:
```python
from depthai_sdk import OakCamera

with OakCamera() as oak:
    color = oak.create_camera('color', name="color", encode='h264')
    oak.sync([color.out.camera, color.out.encoded], callback=lambda p: print(p))
    oak.start(blocking=True)
```
Also, two outputs from the same stream won't work since they will have the same name, perhaps we could use the name as a prefix/postfix or add distinct ids?
I think it'd be useful to add naming for NNComponent and StereoComponent as well.

**Review by daniilpastukhov — COMMENTED** [review]
LGTM

**Pull request merged**